src/azure-cli/azure/cli/command_modules/acs/_params.py

```diff
@@ -324,9 +324,25 @@ def load_arguments(self, _):
 
     with self.argument_context('openshift create') as c:
         c.argument('name', validator=validate_linux_host_name)
-        c.argument('compute_vm_size', options_list=['--compute-vm-size', '-s'])
-        c.argument('customer_admin_group_id', options_list=['--customer-admin-group-id'])
-        c.argument('workspace_id')
+        c.argument('aad_client_app_id', help='The ID of an Azure Active Directory client application. If not specified, a new Azure Active Directory client is created.')
+        c.argument('aad_client_app_secret', help='The secret of an Azure Active Directory client application.')
+        c.argument('aad_tenant_id', help='The ID of an Azure Active Directory tenant.')
+        c.argument('compute_count', options_list=['--compute-count', '-c'], help='Number of nodes in the OpenShift node pool.')
+        c.argument('compute_vm_size', options_list=['--compute-vm-size', '-s'], help='Size of Virtual Machines to create as OpenShift nodes.')
+        c.argument('customer_admin_group_id',
+                   help='The Object ID of an Azure Active Directory Group that memberships will get synced into the OpenShift group "osa-customer-admins".'
+                        'If not specified, no cluster admin access will be granted.')
+        c.argument('management_subnet_cidr', help='CIDR of subnet used to create PLS needed for management of the cluster. If provided, also set --private-cluster flag.')
+        c.argument('private_cluster', arg_type=get_three_state_flag(), help='Create private Openshift cluster. If this flag is set, also supply --management-subnet-cidr.')
+        c.argument('subnet_prefix', help='The CIDR used on the Subnet into which to deploy the cluster.')
+        c.argument('vnet_peer',
+                   help='Vnet peering is no longer supported during cluster creation, instead it is possible to edit vnet properties after cluster creation')
+        c.argument('vnet_prefix', help='The CIDR used on the VNet into which to deploy the cluster.')
+        c.argument('workspace_id', help='The resource id of an existing Log Analytics Workspace to use for storing monitoring data.')
+
+    with self.argument_context('openshift update') as c:
+        c.argument('refresh_cluster', arg_type=get_three_state_flag(),
+                   help='Allow nodes to be rotated. Use this flag to trigger nodes rotation after DNS settings change.')
```

Comment:
I suggest to use `action='store_true'` instead of three_state_flag since `refresh_cluster` is not a property of the resource. See more context [here](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#registering-flags).

Reply:
@fengzhou-msft thank you for reviewing and improvement suggestions. I have implemented them locally but lately there were few problems with RP that prevented from testing my changes. I will hold off for now working on this PR as we are still waiting for the core python sdk to be updated and then I'll test everything together.

Reply:
@fengzhou-msft we do store this flag on the resource properties: https://github.com/openshift/openshift-azure/blob/3f03631cc8a435afde90b158289613ee4140b07e/pkg/api/admin/types.go#L75
It is not returned in the response payload or in subsequent `az openshift show` commands. should I still use `store_true` rather than `three_state_flag`?

Reply:
If it's a resource property, then `get_three_state_flag` is the right choice.